yamux/src/connection.rs

```diff
@@ -352,6 +356,8 @@ struct Active<T> {
     stream_receiver: mpsc::Receiver<StreamCommand>,
     dropped_streams: Vec<StreamId>,
     pending_frames: VecDeque<Frame<()>>,
+    pending_acks: HashSet<StreamId>,
```

Comment:
Technically speaking, this datastructure duplicateds information, right? The fact whether a stream is pending an ack is already contained in the shared state between stream and connection. Do I understand correctly that this `HashSet` is an optimization such that we don't have to iterate a set of locks on each `poll_new_outbound`?

Reply:
> Do I understand correctly that this `HashSet` is an optimization such that we don't have to iterate a set of locks on each `poll_new_outbound`?

I didn't view it as an optimization. I just thoughts it is easier to understand and easier to implement.

I am not a fan of these "god" data structure `Stream` and `Shared` where only half of the data is used by either end.

Reply:
I guess storing this information in `Stream` would avoid such mistakes.

Reply:
We use a spin lock for the shared data structure so it should be pretty much free to iterate a set of locks.